File: src/main/java/com/airwaves/airwavesweb/util/LocationWrapper.java
package com.airwaves.airwavesweb.util;

import com.airwaves.airwavesweb.datastore.User;
import org.apache.commons.math3.ml.clustering.Clusterable;

public class LocationWrapper implements Clusterable {
    private double[] points;
    private User user;

    public LocationWrapper(User user) {
        this.user = user;
        this.points = new double[]{user.getLatitude(), user.getLongitude()};
    }

    @Override
    public double[] getPoint() {
        return this.points;
    }

    public User getUser() {
        return this.user;
    }
}

File: src/main/kotlin/com/airwaves/airwavesweb/util/LocationWrapper.kt
package com.airwaves.airwavesweb.util

import com.airwaves.airwavesweb.datastore.User
import org.apache.commons.math3.ml.clustering.Clusterable

class LocationWrapper(val user: User) : Clusterable {

    private val points: DoubleArray = doubleArrayOf(user.data.latitude, user.data.longitude)

    override fun getPoint(): DoubleArray = points

}